packages/core/src/dialects/mysql/data-types.ts

```diff
@@ -106,13 +102,26 @@ export class DATE extends BaseTypes.DATE {
     return date.format(format);
   }
 
-  sanitize(value: unknown, options?: { timezone?: string }): unknown {
-    if (isString(value) && options?.timezone) {
-      if (isValidTimeZone(options.timezone)) {
-        return dayjs.tz(value, options.timezone).toDate();
+  toSql() {
+    if (this.options.precision != null) {
+      return `DATETIME(${this.options.precision})`;
+    }
+
+    return 'DATETIME';
+  }
+
+  sanitize(value: unknown): unknown {
+    const timezone = this._getDialect().sequelize.options.timezone;
+    // No need to apply timezone if the timezone is UTC
```

Comment:
this should be handled by applyTimezone. I'm not sure I understand why it has a different implementation. applyTimezone is supposed to handle all of this

Reply:
The dates should be working with this removed and using the logic in the Base.DATE datatype